router.go

```diff
@@ -99,6 +102,7 @@ func (m MapRouter) Register(rs []transport.Procedure) {
 			// since type models like Thrift were not previously required to
 			// specify the encoding of every procedure).
 			m.serviceProcedures[sp] = r
+			m.allServices[r.Service] = true
```

Comment:
Instead of setting the service name here _and_ below, let's just update the map in one place, right after we've validated that name it's non-empty

Reply:
Thanks for pointing out. I had the wrong assumption that process will continue running when `panic()` happens.